pop-api/src/v0/sponsorships/traits.rs

```diff
@@ -0,0 +1,21 @@
+//! Traits that can be used by contracts. Including standard compliant traits.
+
+use super::*;
+
+/// The CanSponsor trait.
+#[ink::trait_definition]
+pub trait CanSponsor {
```

Comment:
Why do we need this? Surely a contract just uses the API functions to register sponsorships?

Reply:
I Included it with the idea that a contract A could move the authentication logic into a separate contract B, which is authorized to call into contract A to register a sponsorship.

Now writing I feel that a better naming would have been `IsSponsor`...

But it is not used in the example anyway, so happy to keep this outside until it becomes interesting.